dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/QAEventRestRepository.java

```diff
@@ -0,0 +1,133 @@
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Item;
+import org.dspace.content.QAEvent;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.qaevent.dao.QAEventsDao;
+import org.dspace.qaevent.service.QAEventService;
+import org.dspace.util.UUIDUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort.Direction;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+/**
+ * Rest repository that handle QA events.
+ *
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ *
+ */
+@Component(QAEventRest.CATEGORY + "." + QAEventRest.NAME)
+public class QAEventRestRepository extends DSpaceRestRepository<QAEventRest, String> {
+
+    final static String ORDER_FIELD = "trust";
+
+    @Autowired
+    private QAEventService qaEventService;
+
+    @Autowired
+    private QAEventsDao qaEventDao;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private ResourcePatch<QAEvent> resourcePatch;
+
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    public QAEventRest findOne(Context context, String id) {
+        QAEvent qaEvent = qaEventService.findEventByEventId(id);
+        if (qaEvent == null) {
+            // check if this request is part of a patch flow
+            qaEvent = (QAEvent) requestService.getCurrentRequest().getAttribute("patchedNotificationEvent");
+            if (qaEvent != null && qaEvent.getEventId().contentEquals(id)) {
+                return converter.toRest(qaEvent, utils.obtainProjection());
+            } else {
+                return null;
+            }
+        }
+        return converter.toRest(qaEvent, utils.obtainProjection());
+    }
+
+    @SearchRestMethod(name = "findByTopic")
+    @PreAuthorize("hasAuthority('ADMIN')")
+    public Page<QAEventRest> findByTopic(Context context, @Parameter(value = "topic", required = true) String topic,
+        Pageable pageable) {
+        List<QAEvent> qaEvents = null;
+        long count = 0L;
+        boolean ascending = false;
+        if (pageable.getSort() != null && pageable.getSort().getOrderFor(ORDER_FIELD) != null) {
+            ascending = pageable.getSort().getOrderFor(ORDER_FIELD).getDirection() == Direction.ASC;
+        }
+        qaEvents = qaEventService.findEventsByTopicAndPage(topic,
+            pageable.getOffset(), pageable.getPageSize(), ORDER_FIELD, ascending);
+        count = qaEventService.countEventsByTopic(topic);
+        if (qaEvents == null) {
+            return null;
+        }
+        return converter.toRestPage(qaEvents, pageable, count, utils.obtainProjection());
+    }
+
+    @Override
+    protected void delete(Context context, String eventId) throws AuthorizeException {
```

Comment:
This appears to be missing a `@PreAuthorize` annotation?  I assume it should be `@PreAuthorize("hasAuthority('ADMIN')")` like all the other methods?